ste/sender-blobSymlinks_linux.go

```diff
@@ -0,0 +1,23 @@
+package ste
+
+import (
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+)
+
+func (s *blobSymlinkSender) getExtraProperties() error {
+	if s.jptm.Info().PreservePOSIXProperties {
+		if unixSIP, ok := s.sip.(IUNIXPropertyBearingSourceInfoProvider); ok {
+			// Clone the metadata before we write to it, we shouldn't be writing to the same metadata as every other blob.
```

Comment:
In Golang, maps are handled as pointers, not as primitive, cloneable types.

If I do
```go
a := map[string]int { "a": 1 }
b := a
```

b is not a deep-copy of a. It's just copied the pointer. if I do `b["b"] = 2` and then say `fmt.Println(a["b"])`, you'll get back 2, because they are the same map.

Now, here's the reason we clone it there: AzCopy never deep-copies the metadata map. It's the same map between every blob. If I modify it on one blob, it modifies it on all the rest of them. This causes data races and incorrect properties being persisted.